--- a/modules/launcher/services/Actions.qml
+++ b/modules/launcher/services/Actions.qml
@@ -49,7 +49,7 @@
             } else {
                 list.screenState.launcher = false;
                 if (!SessionManager.exec(command))
-                    Quickshell.execDetached(command);
+                    Quickshell.execDetached(["@systemd-run@", "--user", "--scope", "--slice=app.slice", "--collect", "--quiet", "--expand-environment=no", "--", ...command]);
             }
         }
     }
